client/lib/analytics/ad-tracking.js

```diff
@@ -1971,41 +1700,98 @@ function setupWpcomFloodlightGtag() {
 }
 
 /**
- * Initializes the Facebook pixel.
+ * Returns whether Google Analytics is allowed.
  *
- * When the user is logged in, additional hashed data is being forwarded.
+ * This function returns false if:
+ *
+ * 1. `isGoogleAnalyticsEnabled` is `true`
+ * 2. `ad-tracking` feature is disabled
+ * 3. `Do Not Track` is enabled
+ * 4. the current user could be in the GDPR zone and hasn't consented to tracking
+ * 5. `document.location.href` may contain personally identifiable information
+ *
+ * Note that doNotTrack() and isPiiUrl() can change at any time which is why we do not cache them.
+ *
+ * @returns {Boolean} true if GA is allowed.
  */
-function initFacebook() {
-	if ( user.get() ) {
-		initFacebookAdvancedMatching();
-	} else {
-		window.fbq( 'init', TRACKING_IDS.facebookInit );
+export function isGoogleAnalyticsAllowed() {
+	return (
+		isGoogleAnalyticsEnabled &&
+		config.isEnabled( 'ad-tracking' ) &&
+		! doNotTrack() &&
+		! isPiiUrl() &&
+		mayWeTrackCurrentUserGdpr()
+	);
+}
 
-		/*
-		 * Also initialize the FB pixel for Jetpack.
-		 * However, disable auto-config for this secondary pixel ID.
-		 * See: <https://developers.facebook.com/docs/facebook-pixel/api-reference#automatic-configuration>
-		 */
-		window.fbq( 'set', 'autoConfig', false, TRACKING_IDS.facebookJetpackInit );
-		window.fbq( 'init', TRACKING_IDS.facebookJetpackInit );
-	}
+/**
+ * Fires Google Analytics page view event
+ *
+ * @param {String} urlPath The path of the current page
+ * @param {String} pageTitle The title of the current page
+ */
+export function fireGoogleAnalyticsPageView( urlPath, pageTitle ) {
+	window.gtag( 'config', TRACKING_IDS.wpcomGoogleAnalyticsGtag, {
```

Comment:
I think we need to have all of the configuration properties present when this occurs also; i.e., we need to copy the data from initialization when tracking a pageview. Why? Because while testing I observed this pageview tracking isn't going through as a POST request (i.e., this isn't using the Beacon API as expected) and it is missing `cd3` in the network request, which is supposed to track `client_id`. In other words, everything that was established during Gtag initialization is lost.

Reply:
Thank you! Fixed.